fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/RecordMetaDataBuilderTest.java

```diff
@@ -339,4 +340,33 @@ public void noPrimaryKey() {
                 RecordMetaData.newBuilder().setRecords(TestRecordsNoPrimaryKeyProto.getDescriptor()).getRecordMetaData());
         assertEquals("Record type MyNoPrimaryKeyRecord must have a primary key", e.getMessage());
     }
+
+    @Test
+    public void updateRecords() {
+        RecordMetaDataBuilder builder = RecordMetaData.newBuilder().setRecords(TestRecords1Proto.getDescriptor());
+        final int prevVersion = builder.getVersion();
+        assertNotNull(builder.getRecordType("MySimpleRecord"));
+        assertSame(builder.getRecordType("MySimpleRecord").getDescriptor().getFile(), TestRecords1Proto.getDescriptor());
+        assertNotNull(builder.getRecordType("MyOtherRecord"));
+        builder.updateRecords(TestRecords1EvolvedProto.getDescriptor(), true);
+        RecordMetaData recordMetaData = builder.build(true);
+        assertNotNull(recordMetaData.getRecordType("MySimpleRecord"));
+        assertSame(builder.getRecordType("MySimpleRecord").getDescriptor().getFile(), TestRecords1EvolvedProto.getDescriptor());
+        assertNotNull(recordMetaData.getRecordType("MyOtherRecord"));
+        assertSame(builder.getRecordType("MyOtherRecord").getDescriptor().getFile(), TestRecords1EvolvedProto.getDescriptor());
+        assertNotNull(recordMetaData.getRecordType("AnotherRecord"));
+        assertSame(builder.getRecordType("AnotherRecord").getDescriptor().getFile(), TestRecords1EvolvedProto.getDescriptor());
+        assertSame(builder.getRecordType("AnotherRecord").getSinceVersion(), builder.getVersion());
+        assertTrue(builder.getRecordType("AnotherRecord").getSinceVersion() > prevVersion);
```

Comment:
```suggestion
        assertThat(builder.getRecordType("AnotherRecord").getSinceVersion(), greaterThan(prevVersion));
```

Accepting this suggestion will involve importing the right matcher.